### PR TITLE
feat: warning text for non-numeric answers

### PR DESCRIPTION
**Pull request by connorhaugh**
We now have a form control feedback for when numeric ANSWERS (not answer ranges) have non-numeric answers.

Screenshot 2023-04-24 at 9 25 04 AM

### CODE REVIEW

File: src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx [hunk 1]
```diff
@@ -39,6 +39,14 @@ export const AnswerOption = ({
   const setUnselectedFeedback = hooks.setUnselectedFeedback({ answer, hasSingleAnswer, dispatch });
   const { isFeedbackVisible, toggleFeedback } = hooks.useFeedback(answer);
 
+  const isNumeric = (str) => {
+    if (typeof str !== 'string') { return false; } // we only process strings!
+    // use type coercion to parse the _entirety_ of the string (`parseFloat` alone does not do this)...
+    return !Number.isNaN(str)
```
[the remark this answers is not in the document]
Comment (connorhaugh) [reply src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx:45]: Bad AC is afoot here.

> AC: create an error message for course author that reads “Warning: there was a problem with the staff answer to this problem” when course author does not enter a real number as the correct answer for numerical input problem.

File: src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx [hunk 2]
```diff
@@ -63,6 +71,31 @@ export const AnswerOption = ({
...
+      return (
+        <Form.Group
+          isInvalid={!isNumeric(answer.title) && !!answer.title}
+        >
+          {
+            !isNumeric(answer.title) && !!answer.title
+&& (
+<Form.Control.Feedback type="invalid">
+  <FormattedMessage {...messages.notANumberHelperText} />
+</Form.Control.Feedback>
```
[the remark this answers is not in the document]
Comment (KristinAoki) [reply src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx:84]: That makes sense, feel free to leave it as is. Could you try to figure out why the "x" icon is misaligned with the error text?

File: src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (connorhaugh) [reply src/editors/containers/ProblemEditor/components/EditProblemView/AnswerWidget/AnswerOption.jsx:45]: Following up with brad.

**Comment by connorhaugh** [comment]
Closing this as we moved the ticket back to the backlog for a robust set of ACs from product, which will be very different from the solution proposed here.

**Pull request closed without merging**